File: tests/example-frames/vertices-1.c
#include "ufbx.h"
#include <stdio.h>
#include <stdint.h>

void begin_polygon()
{
    printf("begin polygon\n");
}

void polygon_corner(ufbx_vec3 position, ufbx_vec3 normal)
{
    printf("  position (%.2f, %.2f, %.2f) normal (%.2f, %.2f, %.2f)\n",
        position.x, position.y, position.z,
        normal.x, normal.y, normal.z);
}

void end_polygon()
{
    printf("end polygon\n");
}

// -- EXAMPLE_SOURCE --

int main(int argc, char **argv)
{
    ufbx_scene *scene = ufbx_load_file("my_scene.fbx", NULL, NULL);
    assert(scene);

    ufbx_node *node = ufbx_find_node(scene, "Cube");
    assert(node && node->mesh);

    list_vertices(node->mesh);

    return 0;
}

File: tests/example-frames/vertices-1.rs
use ufbx;

fn begin_polygon() {
    println!("begin polygon");
}

fn polygon_corner(position: ufbx::Vec3, normal: ufbx::Vec3) {
    println!("  position ({:.2}, {:.2}, {:.2}) normal ({:.2}, {:.2}, {:.2})",
        position.x, position.y, position.z,
        normal.x, normal.y, normal.z);
}

fn end_polygon() {
    println!("end polygon");
}

// -- EXAMPLE_SOURCE --

fn main() {
    let scene = ufbx::load_file("my_scene.fbx", ufbx::LoadOpts::default()).unwrap();
    let node = scene.find_node("Cube").unwrap();
    list_vertices(node.mesh.as_ref().unwrap());
}
